File: modelagem-sistemas/uml-poo/Exemplo Aula/Exemplo_Conta_abstract/src/ContaPoupanca.java

public class ContaPoupanca extends Conta {
	
	private float diaDeposito;
	
	public float getDiaDeposito() {
		return diaDeposito;
	}

	public void setDiaDeposito(float diaDeposito) {
		this.diaDeposito = diaDeposito;
	}

	public float verLucro() {
		return getSaldo();
	}
	
}

File: modelagem-sistemas/uml-poo/Aula 1/algoritmo/php/Intro POO/poo/ContaPoupanca.php
<?php
    require 'Cliente.php';

    class ContaPoupanca extends Conta {
        
        private int $_diaDeposito;

        function __construct($agencia, $numeroConta, $diaDeposito)
        {
            $this->Agencia = $agencia;
            $this->NumeroConta = $numeroConta;
            $this->_diaDeposito = $diaDeposito;
        }

        function setDiaDeposito(int $diaDeposito)
        {
            $multiploCinco = $diaDeposito % 5; 

            if ($diaDeposito < 30 && 
                ($multiploCinco == 0)) {
                $this->_diaDeposito = $diaDeposito;
            }
        }

        function VerLucro()
        {
        }

    }
?>